src/test/regress/expected/multi_cluster_management.out

```diff
@@ -323,6 +324,14 @@ SELECT shardid, shardstate, nodename, nodeport FROM pg_dist_shard_placement WHER
 
 CREATE TABLE cluster_management_test_colocated (col_1 text, col_2 int);
 SELECT create_distributed_table('cluster_management_test_colocated', 'col_1', 'hash', colocate_with=>'cluster_management_test');
+WARNING:  could not find any shard placements for shardId 1220017
+WARNING:  could not find any shard placements for shardId 1220019
+WARNING:  could not find any shard placements for shardId 1220021
+WARNING:  could not find any shard placements for shardId 1220023
+WARNING:  could not find any shard placements for shardId 1220025
+WARNING:  could not find any shard placements for shardId 1220027
+WARNING:  could not find any shard placements for shardId 1220029
+WARNING:  could not find any shard placements for shardId 1220031
```

Comment:
what happened here?

Reply:
Several lines earlier the test marks some placements for deletion. All the new colocated shards are somehow not marked as active, and hence this error.

I do not see why the states of the new co-located shards are different than before. Looking into it

```
#0  ShardPlacementList (shardId=shardId@entry=102043) at metadata/metadata_cache.c:829
#1  0x00007fd6b8845773 in ActiveShardPlacementList (shardId=shardId@entry=102043) at metadata/metadata_utility.c:1235
#2  0x00007fd6b8842fc0 in ShardListInsertCommand (shardIntervalList=0x556518816260) at metadata/metadata_sync.c:757
#3  0x00007fd6b8843597 in GetDistributedTableDDLEvents (relationId=relationId@entry=16926) at metadata/metadata_sync.c:524
#4  0x00007fd6b8843e5d in CreateTableMetadataOnWorkers (relationId=relationId@entry=16926) at metadata/metadata_sync.c:1401
#5  0x00007fd6b8800d7f in CreateDistributedTable (relationId=relationId@entry=16926, distributionColumn=distributionColumn@entry=0x556518850090, distributionMethod=<optimized out>, shardCount=shardCount@entry=16, shardCountIsStrict=shardCountIsStrict@entry=false, colocateWithTableName=colocateWithTableName@entry=0x556518850058 "cluster_management_test", viaDeprecatedAPI=false) at commands/create_distributed_table.c:504
#6  0x00007fd6b8801052 in create_distributed_table (fcinfo=<optimized out>) at commands/create_distributed_table.c:258
#7  0x00005565177cf3f3 in ExecInterpExpr (state=0x55651886a738, econtext=0x55651886a428, isnull=0x7ffe338f94ef) at execExprInterp.c:625
#8  0x00005565177cc37b in ExecInterpExprStillValid (state=0x55651886a738, econtext=0x55651886a428, isNull=0x7ffe338f94ef) at execExprInterp.c:1778
#9  0x00005565177ffc6c in ExecEvalExprSwitchContext (isNull=0x7ffe338f94ef, econtext=0x55651886a428, state=0x55651886a738) at ../../../src/include/executor/executor.h:310
#10 ExecProject (projInfo=0x55651886a730) at ../../../src/include/executor/executor.h:344
#11 ExecResult (pstate=<optimized out>) at nodeResult.c:136
#12 0x00005565177da930 in ExecProcNodeFirst (node=0x55651886a310) at execProcnode.c:445
#13 0x00005565177d316b in ExecProcNode (node=0x55651886a310) at ../../../src/include/executor/executor.h:242
#14 ExecutePlan (estate=estate@entry=0x55651886a0b8, planstate=0x55651886a310, use_parallel_mode=<optimized out>, operation=operation@entry=CMD_SELECT, sendTuples=sendTuples@entry=true, numberTuples=numberTuples@entry=0, direction=ForwardScanDirection, dest=0x5565187def08, execute_once=true) at execMain.c:1632
#15 0x00005565177d3de1 in standard_ExecutorRun (queryDesc=queryDesc@entry=0x556518836118, direction=direction@entry=ForwardScanDirection, count=count@entry=0, execute_once=execute_once@entry=true) at execMain.c:350
#16 0x00007fd6b8839a79 in CitusExecutorRun (queryDesc=0x556518836118, direction=ForwardScanDirection, count=0, execute_once=<optimized out>) at executor/multi_executor.c:214
#17 0x00007fd6bb45538b in pgss_ExecutorRun (queryDesc=0x556518836118, direction=ForwardScanDirection, count=0, execute_once=<optimized out>) at pg_stat_statements.c:891
#18 0x00005565177d3e92 in ExecutorRun (queryDesc=queryDesc@entry=0x556518836118, direction=direction@entry=ForwardScanDirection, count=count@entry=0, execute_once=<optimized out>) at execMain.c:292
#19 0x00005565179525b6 in PortalRunSelect (portal=portal@entry=0x55651864bd08, forward=forward@entry=true, count=0, count@entry=9223372036854775807, dest=dest@entry=0x5565187def08) at pquery.c:929
#20 0x0000556517953c80 in PortalRun (portal=portal@entry=0x55651864bd08, count=count@entry=9223372036854775807, isTopLevel=isTopLevel@entry=true, run_once=run_once@entry=true, dest=dest@entry=0x5565187def08, altdest=altdest@entry=0x5565187def08, completionTag=0x7ffe338f9a50 "") at pquery.c:770
#21 0x000055651794ffe2 in exec_simple_query (query_string=query_string@entry=0x55651857b478 "SELECT create_distributed_table('cluster_management_test_colocated', 'col_1', 'hash', colocate_with=>'cluster_management_test');") at postgres.c:1215
#22 0x0000556517951e91 in PostgresMain (argc=<optimized out>, argv=argv@entry=0x556518611d98, dbname=<optimized out>, username=<optimized out>) at postgres.c:4261
#23 0x00005565178c689f in BackendRun (port=port@entry=0x55651860f9b0) at postmaster.c:4484
#24 0x00005565178c9b6b in BackendStartup (port=port@entry=0x55651860f9b0) at postmaster.c:4167
#25 0x00005565178c9d93 in ServerLoop () at postmaster.c:1725
#26 0x00005565178cb303 in PostmasterMain (argc=5, argv=<optimized out>) at postmaster.c:1398
#27 0x0000556517828915 in main (argc=5, argv=0x556518576480) at main.c:228
```

Reply:
any update?

Reply:
While working on test cases, I unintentionally covered a scenario that was not covered before. This is what happens when we mark some shards of a distributed table for deletion, and later on attempt to distribute a colocated table. We do not create any new shards that will be colocated with the shards that are marked for deletion, and hence the error.

On `CreateColocatedShards(..)` we skip over all the shards that are marked for deletion https://github.com/citusdata/citus/blob/924959fdb1d05d9876ecf7436f148f7c3b4062af/src/backend/distributed/operations/create_shards.c#L296-L301

This change was introduced in #3408 and the new test outputs IMO actually capture the expected behavior.

Reply:
Is there any chance for uses to see this warning? Is that something we could fix? (perhaps separately?)

Reply:
I think it is not likely to see this error message. I don't think we need to fix something here.

- A power user can manually set the shardstate to 4, and then try to create some colocated tables.
- If we mark a placement as inactive due to some errors in the cluster, Citus may not create colocated shards.

https://github.com/citusdata/citus/blob/d7dd247fb5ae9c3cdcf12b343b111d702d9a5bf8/src/backend/distributed/executor/adaptive_executor.c#L4849-L4854

These are the only cases that I can think of. I think we should have at least one active placements for each shard in the absence of errors.

Reply:
Got it, makes sense. Maybe add a comment to the test on why these warnings are shown.